ppoprf/src/ppoprf.rs

```diff
@@ -68,6 +69,67 @@ impl ProofDLEQ {
     Self { c: chl, s }
   }
 
+  fn new_batch(
+    key: &RistrettoScalar,         //k
+    public_value: &RistrettoPoint, //Y -> B
+    p: &[RistrettoPoint],          //C
+    q: &[RistrettoPoint],          //D
+  ) -> Self {
+    //(M, Z) = ComputeCompositesFast(k, B, C, D)
+    let (m, z) = ProofDLEQ::compute_composites(Some(*key), public_value, p, q);
+    //r = G.RandomScalar()
+    let mut csprng = OsRng;
+    let r = RistrettoScalar::random(&mut csprng);
+    //t2 = r * A
+    let t2 = r * RISTRETTO_BASEPOINT_POINT;
+    //t3 = r * M
+    let t3 = r * m;
+
+    /*challengeTranscript =
+    I2OSP(len(Bm), 2) || Bm ||
+    I2OSP(len(a0), 2) || a0 ||
+    I2OSP(len(a1), 2) || a1 ||
+    I2OSP(len(a2), 2) || a2 ||
+    I2OSP(len(a3), 2) || a3 ||
+    "Challenge"*/
+    let mut challenge_transcript = Vec::new();
+    challenge_transcript.extend_from_slice(ProofDLEQ::i2osp(
+      &public_value.compress().as_bytes().len().to_be_bytes(),
```

Comment:
Thanks for adding the `i2osp` functions. I think that's more clear.

FWIW, `curve25519-dalek` [hard codes this length value](https://github.com/dalek-cryptography/curve25519-dalek/blob/3.2.1/src/ristretto.rs#L215), but doesn't export the constant for reference. We define `ppoprf::COMPRESED_POINT_LENGTH` to compensate, so you could use that here if you prefer.

However, I expect the compiler can see through the `.len()` call and evaluate this as a constant, hopefully also avoiding the duplicate `.compress()` call, which isn't trivial. So in the absence of evidence for that, one way or the other, I think it's fine to use whichever version you prefer for readability.